src/wallet/rpcwallet.cpp

```diff
@@ -3208,6 +3208,10 @@ void FundTransaction(CWallet* const pwallet, CMutableTransaction& tx, CAmount& f
     if (!pwallet->FundTransaction(tx, fee_out, change_position, strFailReason, lockUnspents, setSubtractFeeFromOutputs, coinControl)) {
         throw JSONRPCError(RPC_WALLET_ERROR, strFailReason);
     }
+
+    if (fee_out > pwallet->m_default_max_tx_fee) {
+        throw JSONRPCTransactionError(TransactionError::MAX_FEE_EXCEEDED);
+    }
```

Comment:
I don't mind having a belt-and-suspenders check in the rpc code (similar to the GUI code), but ultimately the check should happen in the wallet.

In fact the wallet already does this check in a hacky way (`Transaction too large`...). So I'd prefer if you removed the hacky check and replaced it with a check against m_default_max_tx_fee

Reply:
It indeed makes sense for the wallet to do this check. Will look into that.

Reply:
Done